xen/drivers/passthrough/arm/ipmmu-vmsa.c

```diff
@@ -288,14 +484,28 @@ static void ipmmu_ctx_write(struct ipmmu_vmsa_domain *domain, unsigned int reg,
 	ipmmu_write(domain->root, domain->context_id * IM_CTX_SIZE + reg, data);
 }
 
-static void ipmmu_ctx_write2(struct ipmmu_vmsa_domain *domain, unsigned int reg,
+/* Xen: Write the context for cache IPMMU only. */
+static void ipmmu_ctx_write1(struct ipmmu_vmsa_domain *domain, unsigned int reg,
```

Comment:
ipmmu_ctx_write2/ipmmu_ctx_write/ipmmu_ctx_read/ipmmu_read/ipmmu_write are not inline.
So is new helper. At least for now, while I am trying to follow Linux code.